src/functions.jl

```diff
@@ -7,56 +7,75 @@
 #-------------------------------------------------------------------------------
 
 """
-    move_exchange!(solution::Solution, i::Int, j::Int, instance::Instance)
+    move_exchange!(solution::Solution, car_pos_a::Int, car_pos_b::Int, instance::Instance)
 
-Interverts the car `i` with the car `j` in `solution.sequence`. Updates
+Interverts the car `car_pos_a` with the car `car_pos_b` in `solution.sequence`. Updates
 `solution.M1`, `solution.M2` and `solution.M3`.
 """
-function move_exchange!(solution::Solution, i::Int, j::Int, instance::Instance)
-    solution.sequence[i], solution.sequence[j] = solution.sequence[j], solution.sequence[i]
-    update_matrices!(solution, solution.n, instance)
-    # for k in 1:(instance.nb_HPRC + instance.nb_LPRC)
-    #     if xor(instance.RC_flag[i, k], instance.RC_flag[j, k])
-    #         plusminusone = 1
-    #         if instance.RC_flag[i, k]
-    #             plusminusone = -1
-    #         end
-    #         pos1 = i - (instance.RC_q[k] - 1)
-    #         if pos1 < 1
-    #             pos1 = 1
-    #         end
-    #         for l in pos1:i
-    #             solution.M1[k, l] += plusminusone
-    #         end
-    #         pos2 = j - (instance.RC_q[k] - 1)
-    #         if pos2 < 1
-    #             pos2 = 1
-    #         end
-    #         for l in pos2:j
-    #             solution.M1[k, l] -= plusminusone
-    #         end
-    #         pos1 -= 1
-    #         if pos1 < 1
-    #             pos1 = 1
-    #         end
-    #         cpt1 = solution.M2[k, pos1]
-    #         cpt2 = solution.M3[k, pos1]
-    #         for l in pos1:solution.n
-    #             if solution.M1[k, l] > instance.RC_p[k]
-    #                 cpt1 += 1
-    #                 solution.M2[k, l] = cpt1
-    #             else
-    #                 solution.M2[k, l] = cpt1
-    #             end
-    #             if solution.M1[k, l] >= instance.RC_p[k]
-    #                 cpt2 += 1
-    #                 solution.M3[k, l] = cpt2
-    #             else
-    #                 solution.M3[k, l] = cpt2
-    #             end
-    #         end
-    #     end
-    # end
+function move_exchange!(solution::Solution, car_pos_a::Int, car_pos_b::Int, instance::Instance)
+    if car_pos_a > car_pos_b
+        return move_exchange!(solution, car_pos_b, car_pos_a, instance)
+    end
+    if car_pos_a == car_pos_b
+        return solution
+    end
+    solution.sequence[car_pos_a], solution.sequence[car_pos_b] = solution.sequence[car_pos_b], solution.sequence[car_pos_a]
+    #update_matrices!(solution, solution.n, instance)
+    car_a = solution.sequence[car_pos_a]
+    car_b = solution.sequence[car_pos_b]
+    for option in 1:(instance.nb_HPRC + instance.nb_LPRC)
+        if xor(instance.RC_flag[car_a, option], instance.RC_flag[car_b, option])
+            plusminusone = 1
+            if instance.RC_flag[car_a, option]
+                plusminusone = -1
```

Comment:
`car_a` is the new car at `car_pos_a`. So if it has the flag, it means all sequences reaching it will be increase by one for this option.

Reply:
```
solution.sequence[car_pos_a], solution.sequence[car_pos_b] = solution.sequence[car_pos_b], solution.sequence[car_pos_a]

car_a = solution.sequence[car_pos_a]
car_b = solution.sequence[car_pos_b]`
```

You can also change the order here and say `car_a` was the car at `car_pos_a` before the swap